tests/upgrades/scenario_workers.py

```diff
@@ -0,0 +1,58 @@
+import json
+import os
+
+import pytest
+from automation_tools.satellite6.hammer import set_hammer_config
+from fabric.api import env
+
+import robottelo
+from robottelo.config import settings
+
+
+_json_file = 'upgrade_workers.json'
+
+
+def save_worker_id(test_name, worker_id):
+    data = {}
+    if os.path.exists(_json_file):
```

Comment:
I'd encourage you to start using pathlib to handle these file interactions.

```
from pathlib import Path

json_file = Path(_json_file)
if json_file.exists():
    data = json.load(json_file.read_text())
data.update({(test_name.split('[')[0] if '[' in test_name else test_name): worker_id})
json_file.write_text(json.dump(data))
```

Reply:
This code doesnt works for me ! Looking ..